spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala

```diff
@@ -0,0 +1,152 @@
+package com.datastax.spark.connector.rdd
+
+import com.datastax.driver.core.Session
+import com.datastax.spark.connector._
+import com.datastax.spark.connector.cql._
+import com.datastax.spark.connector.metrics.InputMetricsUpdater
+import com.datastax.spark.connector.rdd.reader._
+import com.datastax.spark.connector.util.CountingIterator
+import com.datastax.spark.connector.writer._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.{Partition, TaskContext}
+
+import scala.reflect.ClassTag
+
+// O[ld] Is the type of the left Side RDD, N[ew] the type of the right hand side Results
+/**
+ * An RDD that will do a selecting join between `prev:RDD` and the specified Cassandra Table
+ * This will perform individual selects to retrieve the rows from Cassandra and will take
+ * advantage of RDD's that have been partitioned with the [[com.datastax.spark.connector.rdd.partitioner.ReplicaPartitioner]]
+ */
+class CassandraJoinRDD[O, N] private[connector](prev: RDD[O],
```

Comment:
O for a type parameter is not a very fortunate choice. It is easy to mistake it for 0 (zero).